amundsen_application/static/js/components/common/ResourceList/index.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+import ResourceListItem from 'components/common/ResourceListItem';
+import { Resource } from 'interfaces';
+import { ITEMS_PER_PAGE, PAGINATION_PAGE_RANGE } from './constants';
+
+
+export interface ResourceListProps {
+  items: Resource[];
+  source: string;
+
+  // The following props are only used if 'paginate' is set to true
+  paginate?: boolean;
+  activePage?: number;
+  isFullList?: boolean;
+  itemsCount?: number;
+  itemsPerPage?: number;
+  onPagination?: (pageNumber: number) => void;
+}
+
+interface ResourceListState {
+  activePage: number;
+}
+
+class ResourceList extends React.Component<ResourceListProps, ResourceListState> {
+  public static defaultProps: Partial<ResourceListProps> = {
+    paginate: false,
+    activePage: 0,
+    itemsPerPage: ITEMS_PER_PAGE,
+    isFullList: true,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { activePage: this.props.activePage };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.activePage !== prevProps.activePage) {
+      this.setState({ activePage: this.props.activePage });
+    }
+  }
+
+  onPagination = (rawPageNum: number) => {
+    const activePage = rawPageNum - 1;
+    if (this.props.onPagination !== undefined) {
+      this.props.onPagination(activePage);
+    } else {
+      this.setState({ activePage });
+    }
+  };
+
+
+  render() {
+    const { isFullList, items, itemsPerPage, paginate, source } = this.props;
+    const itemsCount = this.props.itemsCount || items.length;
```

Comment:
Why is `items.length` not sufficient?

Reply:
For `SearchPage` we `items` is an array of at most 10 items, when there might be 3000 items to paginate over.

Reply:
I see. So if I understand correctly we have ~3 different cases for the ways we can display a ResourceList:
1. Total items given, all items rendered with no pagination (some future use case)
2. Total items given, slice of items rendered with pagination (MyBookmarks)
3. Slice of items given with total number of items, slice of items rendered with pagination (search results)

And what's confusing me a bit is that the flexibility is achieved by creating a set of optional properties, and different combinations of those properties render the desired results. I don't want to block progress if this might take too long to address but now I'm leaning towards something like:
`paginationType?:  valid values "none" (default), "fullList", "partialList"`
`fullListOptions?: { all the props needed to render this case }`
`partialListOptions?: { all the props needed to render this case }`
And then unfortunately all the nested props would have to be required due to how React handles nested default props. 

The names in my examples can probably be better, and what I've proposed may need to be tweaked in some way to account for all the use cases, but my thought process here is to see how we can make this component flexible yet easy to work with and less prone to bugs. 

It feels like if we set up our properties in a slightly less consolidated way, it will be easy to place our logic in 3 clear chunks for each case and render accordingly. We would lose some consolidation and be forced to have some nested props of the same name on the different options objects, but it will be clearer for the next person who comes along what needs to be modified if they want to make a change for a particular case. I'm fine tabling this conversation and filing a ticket.